### PR TITLE
Improve alteration of indexes.

### PR DESCRIPTION
**Pull request by mood1994**
SEQUOIASQLMAINSTREAM-242
1. No longer use the first field as the sharding key, if there is no primary key and unique key.
2. Renaming indexes is not supported when inplace.
3. If it's a change of the secondary attributes, don't recreate indexes.

### CODE REVIEW

File: ha_sdb.cc [hunk 1]
```diff
@@ -1686,22 +1718,45 @@ bool ha_sdb::inplace_alter_table(TABLE *altered_table, Alter_inplace_info *ha_al
...
+      ha_alter_info->handler_flags & INPLACE_ONLINE_ADDIDX) {
+    KEY *drop_key = NULL;
+    KEY *add_key = NULL;
+    for (uint i = 0; i < ha_alter_info->index_drop_count; i++) {
+      drop_key = ha_alter_info->index_drop_buffer[i];
+      for (uint j = 0; j < ha_alter_info->index_add_count; j++) {
+        add_key =
+            &ha_alter_info->key_info_buffer[ha_alter_info->index_add_buffer[j]];
+        if (sdb_is_same_index(drop_key, add_key)) {
+          ignored_keys.push_back(drop_key->name);
```
Comment (davidli2010) [inline ha_sdb.cc]: ignored_keys can be more effective by using Bitmap.

File: sdb_idx.cc [hunk 2]
```diff
@@ -627,3 +621,32 @@ int sdb_create_condition_from_key(TABLE *table, KEY *key_info,
...
+
+my_bool sdb_is_same_index(const KEY *a, const KEY *b) {
+  my_bool rs = false;
+  const KEY_PART_INFO *key_part_a = NULL;
+  const KEY_PART_INFO *key_part_b = NULL;
+
+  if (strcmp(a->name, b->name) != 0 ||
+      a->user_defined_key_parts != b->user_defined_key_parts ||
+      (a->flags & HA_NOSAME) != (b->flags & HA_NOSAME)) {
+    goto error;
```
[the remark this answers is not in the document]
Comment (mood1994) [reply sdb_idx.cc]: Ok.

File: sdb_idx.cc [hunk 3]
```diff
@@ -627,3 +621,32 @@ int sdb_create_condition_from_key(TABLE *table, KEY *key_info,
...
+  }
+
+  key_part_a = a->key_part;
+  key_part_b = b->key_part;
+  for (uint i = 0; i < a->user_defined_key_parts; ++i) {
+    if (strcmp(key_part_a->field->field_name, key_part_b->field->field_name) !=
+        0) {
+      goto error;
```
[the remark this answers is not in the document]
Comment (mood1994) [reply sdb_idx.cc]: Ok.

File: ha_sdb.cc (hunk 1 above)
Comment (mood1994) [reply ha_sdb.cc]: Ok. I did it in recent commits.

**Pull request merged**